shipment_advice_bill_auto_complete/models/account_move.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from collections import defaultdict
+
+from odoo import _, api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    shipment_advice_auto_complete_id = fields.Many2one("shipment.advice", store=False)
+
+    @api.onchange("shipment_advice_auto_complete_id")
+    def _onchange_shipment_auto_complete(self):
+        shipment = self.shipment_advice_auto_complete_id
+        self.shipment_advice_auto_complete_id = False
+        if not shipment.shipment_type == "incoming":
+            return
+        if not shipment.supplier_ids:
+            return
+        if len(shipment.supplier_ids) > 1:
+            return {
+                "warning": {
+                    "title": _("Shipment with multiple suppliers"),
+                    "message": _(
+                        "There are multiple suppliers on the shipment advice."
+                        "It can not be used to generate the vendor bill."
+                    ),
+                    "type": "notification",
+                },
+            }
+        if not self.partner_id:
+            self.partner_id = shipment.supplier_ids
+        if self.partner_id not in shipment.supplier_ids:
+            return
+
+        # Until the bill is saved, it is possible to select the same shipment multiple time
+        # In this case remove all invoicing lines related to that shipment
+        self.line_ids = self.line_ids.filtered(
+            lambda line: line.shipment_advice_id != shipment
+        )
+
+        account_move_lines = self.env["account.move.line"]
+        sequence = max(self.line_ids.mapped("sequence")) + 1 if self.line_ids else 10
+
+        # FIXME should use the loaded_move_line_ids but does not
+        #       seem to work at least with incoming shipment
+        # move_lines = shipment.loaded_move_line_ids
```

Comment:
To me it looks like the shipment advice id is not properly updated on the move lines.

Reply:
So could be an issue with the `shipment_advice` module.
Also the naming of the fields is a confusing. Because they have been named for outgoing shipment and they do not make sense for incoming ones.